includes/gateways/stripe/includes/class-give-stripe-payment-intent.php

```diff
@@ -56,21 +57,23 @@ public function create( $args ) {
 					$args,
 					give_stripe_get_connected_account_options()
 				);
-			} catch ( Exception $e ) {
-
-				give_record_gateway_error(
-					__( 'Stripe Payment Intent Error', 'give' ),
-					sprintf(
-						/* translators: %s Exception Error Message */
-						__( 'Unable to create a payment intent. Details: %s', 'give' ),
-						$e->getMessage()
-					)
-				);
-
-				give_set_error( 'stripe_payment_intent_error', __( 'Error creating payment intent with Stripe. Please try again.', 'give' ) );
-				return false;
-			} // End try().
-		}
+            } catch (Exception $e) {
+                give_record_gateway_error(
+                    __('Stripe Payment Intent Error', 'give'),
+                    sprintf(
+                    /* translators: %s Exception Error Message */
+                        __('Unable to create a payment intent. Details: %s', 'give'),
+                        $e->getMessage()
+                    )
+                );
+
+                give_set_error('stripe_payment_intent_error',
+                    $e->getMessage()
+                );
```

Comment:
I don't really recommend doing this generically with exception messages. I can see below that you're writing the message in the exception itself and displaying it here, but this `catch (Exception $e)` will catch _any_ exception, and then we're risking displaying very technical and confusing messages to the donor.

In general, I try to make exception messages clear to the developer who stumbles on it to help them debug the issue. So I'd probably leave the exception message as it is, and update the message here.

Reply:
Resolved: https://github.com/impress-org/givewp/pull/6787/commits/c7b9ef284c1aaffa97dc503f5b75fd7c8d3a5e0c